### PR TITLE
Update trainer.py

### PR DESCRIPTION
**Pull request by lorenzoFabbri**
Moved `optimizer.step()` before `lr_scheduler.step()`.

### CODE REVIEW

**Comment by williamFalcon** [comment]
each step in a scheduler is for an epoch. this change makes each step equal to each batch which doesn’t work.
